docs/http_api.md

```diff
@@ -0,0 +1,114 @@
+# SQLD HTTP API
+
+This is the documentation for the sqld HTTP API.
+
+## Usage
+
+### The `Value` type
+
+The `Value` type represents an SQLite value. It has 4 variants:
+
+- Text: a UTF-8 encoded string
+- Integer: a 64-bit signed integer
+- Real: a 64-bits floating number
+- Blob: some binary data, encoded in base64
+- Null: the null value.
+
+All these types map to JSON straightforwardly, with two exceptions:
+- Blob: they are represented as an object with { "type": "blob", "base64": /* base64 encoded blob */}
+- Integers: they are represented as string since some languages are not able to represent them in full fidelity.
```

Comment:
How does a client discern the type of an integer vs string if they are both encoded as strings?  Maybe everything should be an object like blob with an explicit type.

Reply:
I'm not sure about this one tbh, on the one hand typing it is unambiguous, but this is also a significant overhead for integers.  Usually the schema is known, so the user can parse to an integer. If it is not know, parsing to an integer will also succeed, but leaves an ambiguity over the type the field is actually supposed to contain. On the other hand, if it quacks like a duck...

Reply:
I'm thinking of introspection/visualization tools that we or the developer might build for themselves.

Another thought: if we make the assumption that all values in a column have the same type, maybe the type info should be in the row spec.  Then it won't clutter up the values so much.

Reply:
With SQLite being dynamically types, that would probably be confusing... 

@BearLemma stock js json serializer handles that very badly 

With that being said maybe it's the duty of the développer so ensure that he can deserialize safely? We, as the provider of the  SDK, can expose a safe API, would that be a reasonable approach?

Reply:
I'd still like to let them know that they're getting something that's not intuitively what they might expect.  If we can tag that row (and perhaps all rows) with type metadata, that would be helpful.

Reply:
I'm proposing that we leave integers as is, so it's unambiguous, and handle overflow at the SDK level.

It's not even mandatory to specify a type in the columns in SQLite, so I'm not sure how useful tagging cols would be. We can add it, but it most likely be null, and when not, not to be trusted